src/combinatorics/simple_graph/fin_subgraph.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2022 Joanna Choules. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joanna Choules
+-/
+import topology.category.Top.limits
+import combinatorics.simple_graph.subgraph
+
+/-!
+# Homomorphisms from finite subgraphs
+
+This file defines the type of finite subgraphs of a `simple_graph` and proves a compactness result
+for homomorphisms to a finite codomain.
+
+## Main statements
+
+* `simple_graph.exists_hom_of_all_finite_homs`: If every finite subgraph of a (possibly infinite)
+  graph `G` has a homomorphism to some finite graph `F`, then there is also a homomorphism `G →g F`.
+
+## Notations
+
+`→fg` is a variant on `→g` where the domain is a finite subgraph of some supergraph `G`.
+
+## Implementation notes
+
+The proof here uses compactness as formulated in `nonempty_sections_of_fintype_inverse_system`. For
+finite subgraphs `G'' ≤ G'`, the inverse system `fin_subgraph_hom_functor` restricts homomorphisms
+`G' →fg F` to domain `G''`.
+-/
+
+universes u v
+variables {V : Type u} {W : Type v} {G : simple_graph V} {F : simple_graph W}
+
```

Comment:
Let's wrap all of your code (after or before `variables` as you prefer) in `namespace simple_graph` so that every declaration ends up in this namespace. Be sure to remove the `simple_graph` qualifiers on the two declarations that include it!